lightstep/sdk/metric/internal/asyncstate/callback.go

```diff
@@ -0,0 +1,102 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package asyncstate // import "github.com/lightstep/otel-launcher-go/lightstep/sdk/metric/internal/asyncstate"
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"go.opentelemetry.io/otel/metric/instrument"
+)
+
+// Callback is the implementation object associated with one
+// asynchronous callback.
+type Callback struct {
+	// function is the user-provided callback function.
+	function func(context.Context)
```

Comment:
Should it be possible for the callback function to return an error? Would that make error handling simpler?

Reply:
This is dictated by the OTel API. Returning errors here just complicates the SDK's life, for there's not much to do. A user could use the `otel.Handle(error)` API to report general trouble.